src/sounds.cpp

```diff
@@ -388,16 +393,790 @@ std::pair<std::vector<tripoint>, std::vector<tripoint>> sounds::get_monster_soun
+            start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+            return;
+        }
+    }
+    int mon_pos = g->mon_at( source );
+    if( mon_pos != -1 ) {
+        monster *monster = g->monster_at( source );
+        std::string monster_id = monster->type->id;
+        if( distance <= 18 ) {
+            if( monster_id == "mon_turret" || monster_id == "mon_secubot" ) {
+                play_variant_sound( "fire_gun", "hk_mp5", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            } else if( monster_id == "mon_turret_rifle" || monster_id == "mon_chickenbot"
+                       || monster_id == "mon_tankbot" ) {
+                play_variant_sound( "fire_gun", "m4a1", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            } else if( monster_id == "mon_turret_bmg" ) {
+                play_variant_sound( "fire_gun", "m2browning", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            } else if( monster_id == "mon_laserturret" ) {
+                play_variant_sound( "fire_gun", "laser_rifle", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            }
+        } else {
+            if( monster_id == "mon_turret" || monster_id == "mon_secubot" ) {
+                play_variant_sound( "fire_gun_distant", "smg", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            } else if( monster_id == "mon_turret_rifle" || monster_id == "mon_chickenbot"
+                       || monster_id == "mon_tankbot" ) {
+                play_variant_sound( "fire_gun_distant", "rifle", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            } else if( monster_id == "mon_turret_bmg" ) {
+                play_variant_sound( "fire_gun_distant", "rifle", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            } else if( monster_id == "mon_laserturret" ) {
+                play_variant_sound( "fire_gun_distant", "laser", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            }
+        }
+    }
+}
+
+void sfx::generate_melee_soundfx( tripoint source, tripoint target, bool hit, bool targ_mon,
+                                  std::string material ) {
+    sound_thread * out = new sound_thread();
```

Comment:
Who is responsible for deleting this object? - "Do you want memory leaks? Because that's how you get memory leaks."

Btw. C++11 (which is already required), has a thread interface. I don't know what compilers support it, but gcc 5.1.0 works fine.

This example works fine, and there are no memory leaks in it:
```C++
#include <thread>
#include <iostream>
#include <string>
#include <chrono>

struct Callable {
    std::string message;
    void operator()() {
        for (int i = 0; i < 5; ++i) {
            std::cout << (message + "\n");
            std::this_thread::sleep_for(std::chrono::milliseconds(10));
        }
    }
};

int main()
{
    std::thread t1( Callable{ "thread 1" } );
    std::thread t2( Callable{ "thread 2" } );
    std::thread t3( Callable{ "thread 3" } );
    t1.join();
    t2.join();
    t3.detach();
    // Important: either `join` or `detach` have to be called
    // before the thread object is deleted.
    std::cout << "Done\n";
}
```
I needed to compile it with `-lpthread` as `g++ -lpthread -std=c++11 test.cpp`